File: backend/src/main/java/com/snackoverflow/toolgether/domain/user/dto/MeInfoResponse.java
package com.snackoverflow.toolgether.domain.user.dto;

import com.snackoverflow.toolgether.domain.user.entity.User;
import lombok.NonNull;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;

public record MeInfoResponse(
        @NonNull Long id,
        @NonNull String nickname,
        String username,
        String profileImage,
        String email,
        @NonNull String phoneNumber,
        @NonNull AddressInfo address,
        @NonNull LocalDateTime createdAt,
        @NonNull Integer score,
        @NonNull Integer credit
) {
    public static MeInfoResponse from(User user) {
        return new MeInfoResponse(
                user.getId(),
                user.getNickname(),
                user.getUsername(),
                user.getProfileImage(),
                user.getEmail(),
                user.getPhoneNumber(),
                AddressInfo.from(user.getAddress()),
                user.getCreatedAt(),
                user.getScore(),
                user.getCredit()
        );
    }
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/user/dto/response/MeInfoResponse.kt
package com.snackoverflow.toolgether.domain.user.dto.response;

import com.snackoverflow.toolgether.domain.user.entity.User;

import java.time.LocalDateTime;

data class MeInfoResponse(
    val id: Long,
    val nickName: String,
    val email: String,
    val profileImage: String?,
    val phoneNumber: String,
    val baseAddress: String,
    val createdAt: LocalDateTime,
    val score: Int,
    val credit: Int
) {
    companion object {
        @JvmStatic
        fun from(user: User): MeInfoResponse {
            return MeInfoResponse(
                user.id!!,
                user.nickname!!,
                user.profileImage!!,
                user.email!!,
                user.phoneNumber!!,
                user.baseAddress!!,
                user.createdAt!!,
                user.score,
                user.credit
            )
        }
    }
}
